Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -316,6 +319,12 @@ theorem testBit_two_pow (n m : ℕ) : testBit (2 ^ n) m = (n = m) := by
     simp [h]
 #align nat.test_bit_two_pow Nat.testBit_two_pow
 
+lemma and_two_pow {n i} : n &&& 2 ^ i = (n.testBit i).toNat * 2 ^ i := by
+  apply eq_of_testBit_eq; intro j
+  rw [mul_comm, testBit_land]
+  cases' h : n.testBit i <;> cases' (ne_or_eq i j) with h1 h1
+  <;> simp [testBit_two_pow_of_ne _, *] at * <;> assumption
```

Comment:
This is a non-terminal simp (more likely, one on each branch) so should be squeezed, or written a different way.

Reply:
I am not sure which pattern had to be avoided, but I interpreted this as adding 'only' and dropping the *. Let me know what else you had in mind.

Reply:
A "non-terminal `simp`" is a `simp` call that has no only *and* is not the last tactic in its goal. Two of your four simps in the revised version didn't need to be squeezed (via `only`), as they were terminal.

I pushed a small cleanup.

Reply:
Impressive. I really need to code more lean. Thank you! The current reviewing experience is outstanding.